target/linux/ramips/dts/mt7620n_hootoo_ht-tm05.dts

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7620n.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "hootoo,ht-tm05", "ralink,mt7620n-soc";
+	model = "HooToo HT-TM05";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		label-mac-device = &ethernet;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "ht-tm05:blue:power";
+			gpios = <&gpio3 0 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		wifi {
+			label = "ht-tm05:green:wifi";
+			gpios = <&gpio2 0 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio2 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	virtual_flash {
+		compatible = "mtd-concat";
+		devices = <&firmware1 &firmware2 &firmware3>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				compatible = "openwrt,okli";
+				label = "firmware";
+				reg = <0x0 0x0>;
+			};
+		};
+	};
+};
+
+&gpio2 {
+	status = "okay";
+};
+
+&gpio3 {
+	status = "okay";
+};
+
+&i2c {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "config";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@50000 {
+				label = "loader";
+				reg = <0x50000 0x1000>;
```

Comment:
I'd avoid partitions < 64k here due to the various problem we (probably) have with 4k sectors on 5.4 and just make this 0x10000 big if there isn't a specific reason against it apart from loosing a little flash space.

Reply:
Related to 4k: there were sysupgrade problems when the recipe set `BLOCKSIZE` to `64k` but didn't used `pad-to` in parallel for the `sysupgrade.bin` recipe.

To the subject: the device has 8 MByte flash, and has a 6 MByte size limitation for kernel + rootfs already.
This 6 MByte limit could be improved later with splitting the tail to the `kernel.bin`.

Reply:
Also here ... just to understand, change this or leave it? I can try changing it, see if it works (like you say @xabolcs, some issues in the past with 64k). The part has a sector size of 4k, so no issues there.

Thanks!

Reply:
OK, got this working - at least I think 🤣. The preference seems to be 64k, so I updated to that, and it all seems to be working. Pushing a new commit, please do look and yell if you disagree. And to make it easier to confirm, here is the serial log,
https://paste.ubuntu.com/p/VQNyW9mVQp/

Thanks!